frontend/src/components/charts/SubmissionsHeatmap.tsx

```diff
@@ -1,20 +1,60 @@
+import { useState } from "react";
 import { useTheme } from "@mui/material/styles/";
-import * as echarts from "echarts";
+import { format } from "date-fns";
 import ReactEcharts from "echarts-for-react";
 
-export const SubmissionsHeatmap = () => {
-  function getVirtualData(year: string) {
-    const date = +echarts.number.parseDate(year + "-01-01");
-    const end = +echarts.number.parseDate(+year + 1 + "-01-01");
-    const dayTime = 3600 * 24 * 1000;
+type SubmissionHeatmapProps = {
+  heatmapData:
+    | {
+        date: Date | string;
+      }[]
+    | undefined;
+};
+
+export const SubmissionsHeatmap = ({ heatmapData }: SubmissionHeatmapProps) => {
+  const theme = useTheme();
+  const [maxSubmissions, setMaxSubmissions] = useState<number>(0);
+
+  const processData = (
+    heatmapData:
+      | {
+          date: Date | string;
+        }[]
+      | undefined,
+  ) => {
+    const dateSubmissionMap = new Map<string, number>();
     const data = [];
-    for (let time = date; time < end; time += dayTime) {
-      // Data element consist of a date and a submission count
-      data.push([new Date(time), Math.floor(Math.random() * 1000)]);
+    if (!heatmapData) {
+      return;
+    } else {
```

Comment:
nit: `else` after `return` is redundant
```suggestion
    }
```